core/sr-primitives/src/generic/checked_extrinsic.rs

```diff
@@ -19,25 +19,33 @@
 
 use crate::traits::{self, Member, SimpleArithmetic, MaybeDisplay};
 use crate::weights::{Weighable, Weight};
+use crate::generic::tip::{Tip, Tippable};
 
 /// Definition of something that the external world might want to say; its
 /// existence implies that it has been checked and is good, particularly with
 /// regards to the signature.
 #[derive(PartialEq, Eq, Clone)]
 #[cfg_attr(feature = "std", derive(Debug))]
-pub struct CheckedExtrinsic<AccountId, Index, Call> {
+pub struct CheckedExtrinsic<AccountId, Index, Call, Balance> {
 	/// Who this purports to be from and the number of extrinsics have come before
 	/// from the same signer, if anyone (note this is not a signature).
 	pub signed: Option<(AccountId, Index)>,
```

Comment:
```suggestion
	pub signed: Option<(AccountId, Index, Tip<Balance>)>,
```

Reply:
it could theoretically be an option, but as long as it's compact encoded, then there's no point since zero compact encoded is a big as `None` encoded.

Reply:
Just make sure `Balance` has a default. For non-tipping chains, `Balance` can be `()`.

Reply:
if it is part of the signature then it is only visible in signed txs which is the point of the current option. Hence, `pub signed: Option<(AccountId, Index, Tip<Balance>)>,` looks ok